crates/starknet-server/src/api/json_rpc/endpoints.rs

```diff
@@ -180,22 +180,47 @@ impl JsonRpcHandler {
                 transaction_type = TransactionType::Declare;
                 Transaction::Declare(crate::api::models::transaction::DeclareTransaction::Version2(
                     DeclareTransactionV2 {
-                        class_hash: declare_v2.class_hash.unwrap(),
+                        class_hash: declare_v2.class_hash.unwrap_or_default(),
                         sender_address: ContractAddressHex(declare_v2.sender_address),
                         nonce: declare_v2.nonce,
                         max_fee: Fee(declare_v2.max_fee),
                         version: Felt::from(2),
-                        transaction_hash: declare_v2.transaction_hash.unwrap(),
+                        transaction_hash: declare_v2.transaction_hash.unwrap_or_default(),
                         signature: declare_v2.signature,
                         compiled_class_hash: declare_v2.compiled_class_hash,
                     },
                 ))
             }
-            starknet_core::transactions::Transaction::DeployAccount(_deploy) => {
-                return Err(error::ApiError::TransactionNotFound);
+            starknet_core::transactions::Transaction::DeployAccount(deploy) => {
+                transaction_type = TransactionType::DeployAccount;
+                Transaction::DeployAccount(
+                    crate::api::models::transaction::DeployAccountTransaction {
+                        transaction_hash: Felt::from(deploy.inner.hash_value().clone()),
+                        max_fee: Fee(deploy.max_fee),
+                        version: deploy.version,
+                        signature: deploy.signature.clone(),
+                        nonce: deploy.nonce,
+                        class_hash: deploy.class_hash().unwrap_or_default(),
+                        contract_address_salt: deploy.contract_address_salt(),
+                        constructor_calldata: deploy.constructor_calldata(),
+                    },
+                )
             }
-            starknet_core::transactions::Transaction::Invoke(_invoke) => {
-                return Err(error::ApiError::TransactionNotFound);
+            starknet_core::transactions::Transaction::Invoke(invoke) => {
+                transaction_type = TransactionType::Invoke;
+                Transaction::Invoke(crate::api::models::transaction::InvokeTransaction::Version1(
+                    InvokeTransactionV1 {
+                        transaction_hash: Felt::from(invoke.inner.hash_value().clone()),
+                        max_fee: Fee(invoke.max_fee),
+                        version: invoke.version,
+                        signature: invoke.signature.clone(),
+                        nonce: invoke.nonce,
+                        calldata: invoke.calldata.clone(),
+                        sender_address: ContractAddressHex(
+                            invoke.sender_address().unwrap_or_default(),
```

Comment:
here the same as the one above, unwrap_or_default operates on Result

Reply:
I removed mapping code and use `Ok(TransactionWithType::try_from(&transaction.inner)?)` in 3df0fb73f32aaf36ef00556c45495078384dcc96